### PR TITLE
feat: ext cluster blob issue

### PR DESCRIPTION
**Pull request by prakash100198**
Previously AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY were not being passed as env variable in common lib, it was set in env var in orch, but this pr aims at setting AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY as env for aws in common lib

### CODE REVIEW

**Review by subhashish-devtron — APPROVED** [review]

File: blob-storage/Bean.go [hunk 1]
```diff
@@ -44,6 +81,34 @@ type AzureBlobConfig struct {
 	AccountKey            string `json:"accountKey"`
 }
 
+func (b *AzureBlobConfig) GetBlobStorageBaseAzureConfig(blobStorageObjectType string) *AzureBlobBaseConfig {
```
Comment (kripanshdevtron) [inline blob-storage/Bean.go:84]: remove duplicacy

File: blob-storage/Bean.go [hunk 2]
```diff
@@ -25,6 +25,43 @@ type BlobStorageS3Config struct {
 	CiArtifactBucketVersioning bool   `json:"ciArtifactBucketVersioning"`
 }
 
+func (b *BlobStorageS3Config) GetBlobStorageBaseS3Config(blobStorageObjectType string) *AwsS3BaseConfig {
```
Comment (kripanshdevtron) [inline blob-storage/Bean.go:28]: remove duplicacy

File: blob-storage/Bean.go [hunk 3]
```diff
@@ -58,6 +123,28 @@ type GcpBlobConfig struct {
 	ArtifactBucketName     string `json:"artifactBucketName"`
 }
 
+func (b *GcpBlobConfig) GetBlobStorageBaseGcpConfig(blobStorageObjectType string) *GcpBlobBaseConfig {
```
Comment (kripanshdevtron) [inline blob-storage/Bean.go:126]: remove duplicacy

File: blob-storage/Bean.go [hunk 4]
```diff
@@ -66,8 +153,11 @@ type GcpBlobBaseConfig struct {
...
-	BLOB_STORAGE_S3                    = "S3"
-	BLOB_STORAGE_GCP                   = "GCP"
-	BLOB_STORAGE_MINIO                 = "MINIO"
+	BLOB_STORAGE_AZURE            BlobStorageType = "AZURE"
+	BLOB_STORAGE_S3                               = "S3"
+	BLOB_STORAGE_GCP                              = "GCP"
+	BLOB_STORAGE_MINIO                            = "MINIO"
+	BlobStorageObjectTypeCache                    = "cache"
+	BlobStorageObjectTypeArtifact                 = "artifact"
+	BlobStorageObjectTypeLog                      = "log"
```
Comment (kripanshdevtron) [inline blob-storage/Bean.go:162]: these 3 are not blobStorageType

File: blob-storage/AwsS3Blob.go [hunk 5]
```diff
@@ -29,7 +29,14 @@ func (impl *AwsS3Blob) UploadBlob(request *BlobStorageRequest, err error) error
 	if s3BaseConfig.Region != "" {
 		cmdArgs = append(cmdArgs, "--region", s3BaseConfig.Region)
 	}
+
 	command := exec.Command("aws", cmdArgs...)
+	if s3BaseConfig.AccessKey != "" && s3BaseConfig.Passkey != "" {
+		command.Env = append(os.Environ(),
+			fmt.Sprintf("AWS_ACCESS_KEY_ID=%s", s3BaseConfig.AccessKey),
```
Comment (subhashish-devtron) [inline blob-storage/AwsS3Blob.go:36]:
```go
func setAWSEnvironmentVariables(config S3Config) {
    if config.AccessKey != "" && config.Passkey != "" {
        os.Setenv("AWS_ACCESS_KEY_ID", config.AccessKey)
        os.Setenv("AWS_SECRET_ACCESS_KEY", config.Passkey)
    }
}
```
cleaner IMO

File: blob-storage/Bean.go [hunk 6]
```diff
@@ -25,6 +25,34 @@ type BlobStorageS3Config struct {
 	CiArtifactBucketVersioning bool   `json:"ciArtifactBucketVersioning"`
 }
 
+func (b *BlobStorageS3Config) GetBlobStorageBaseS3Config(blobStorageObjectType string) *AwsS3BaseConfig {
```
Comment (subhashish-devtron) [inline blob-storage/Bean.go:28]: I don't understand why is this logic here in common lib

File: blob-storage/Bean.go (hunk 2 above)
Comment (prakash100198) [reply blob-storage/Bean.go:28]: done

File: blob-storage/AwsS3Blob.go (hunk 5 above)
Comment (prakash100198) [reply blob-storage/AwsS3Blob.go:36]: done

File: blob-storage/Bean.go (hunk 1 above)
Comment (prakash100198) [reply blob-storage/Bean.go:75]: done

File: blob-storage/Bean.go (hunk 3 above)
Comment (prakash100198) [reply blob-storage/Bean.go:110]: done

File: blob-storage/Bean.go (hunk 4 above)
Comment (prakash100198) [reply blob-storage/Bean.go:143]: that's why I've used BlobStorageObjectTypeCache   "object" here not simply  blobStorageType, if it makes any other sense let's discuss over the naming

**Review by subhashish-devtron — APPROVED** [review]

**Pull request merged**